packages/explorer/src/app/(explorer)/queries/useTableDataQuery.ts

```diff
@@ -51,16 +51,16 @@ export function useTableDataQuery({ table, query }: Props) {
       const schemaKeys = Object.keys(table.schema);
       const result = data.result[0];
       const columnKeys = result[0]
-        .map((columnKey) => {
+        ?.map((columnKey) => {
           const schemaKey = schemaKeys.find((schemaKey) => schemaKey.toLowerCase() === columnKey);
           return schemaKey || columnKey;
         })
         .filter((key) => schemaKeys.includes(key));
-      const rows = result.slice(1).map((row) => Object.fromEntries(columnKeys.map((key, index) => [key, row[index]])));
+      const rows = result.slice(1)?.map((row) => Object.fromEntries(columnKeys.map((key, index) => [key, row[index]])));
 
       return {
-        columns: columnKeys,
-        rows,
+        columns: columnKeys || [],
+        rows: rows || [],
```

Comment:
any way we can improve the types of the above so that we have a better sense for when this happens?

Reply:
nice, `noUncheckedIndexedAccess` is actually very helpful here. Added to the explorer, and fixed all TS errors that that bubbled up. Also adjusted `useTableDataQuery` to reflect better on what should happen with rows data given column keys.